src/Tuura/Plato/Translation.hs

```diff
@@ -1,8 +1,12 @@
 module Tuura.Plato.Translation where
 
 import Data.Char
+import Data.List.Extra
 
-data ValidationResult a = Valid | Invalid [a] [a] [a] deriving Eq
+import Tuura.Concept.Circuit.Basic
+import Tuura.Concept.Circuit.Derived
+
+data ValidationResult a = Valid | Invalid [a] [a] [a] [a] deriving Eq
```

Comment:
I'm not overly sure how we can do that. It makes sense, in that it's invalid. 
I suppose I could create a data type, say `Errors a` which holds these 4 pieces of information?
Or maybe split off inconsistent and undefined initial states into some form of `InitialError` type, which is handled within itself?